pkg/validate/validate.go

```diff
@@ -67,3 +67,42 @@ func NewValidation(
 		return nil, fmt.Errorf("unknown validation type")
 	}
 }
+
+// All runs all given validations and exists at the first technical error
+func All(ctx context.Context, validations ...BuildPath) error {
+	for _, validatation := range validations {
+		if err := validatation.ValidatePath(ctx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// BuildRunFields runs field validations against a BuildRun to detect
+// disallowed field combinations
+func BuildRunFields(buildRun *build.BuildRun) error {
+	if buildRun.Spec.BuildSpec != nil {
+		if buildRun.Spec.BuildRef != nil {
+			return fmt.Errorf("fields 'buildRef' and 'buildSpec' are mutually exclusive")
+		}
+
+		if buildRun.Spec.Output != nil {
+			return fmt.Errorf("cannot use 'output' override and 'buildSpec' simultaneously")
```

Comment:
Looking for feedback from a native speaker. I think "simultaneously" is more meant for two different activities that happen at the some time. Here, "together" might be a better fit.